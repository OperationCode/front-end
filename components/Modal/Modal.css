.ModalCard {
  position: relative;
}

.scrollableContainer {
  margin-top: 2.5rem;
  overflow: auto;
  max-height: 75vh;
}

/* React-Modal Overrides
 * Remove scroll on the body when react-modal is open */
/* stylelint-disable */
:global(.ReactModal__Html--open),
:global(.ReactModal__Body--open) {
  height: 100%;
  overflow: hidden;
}

:global(.ReactModal__Overlay) {
  align-items: center;
  display: flex;
  justify-content: center;
}

:global(.ReactModal__Content) {
  width: 80%;
}
