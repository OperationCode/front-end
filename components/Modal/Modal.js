import React from 'react';
import { node, string, bool, func } from 'prop-types';
import classNames from 'classnames';
import ReactModal from 'react-modal';
import { gtag } from 'common/utils/thirdParty/gtag';
import CardStyles from 'components/Cards/Card/Card.module.css';
import CloseButton from 'components/Buttons/CloseButton/CloseButton';

import ModalStyles from './Modal.module.css';

Modal.propTypes = {
  children: node.isRequired,
  className: string,
  isOpen: bool,
  onRequestClose: func.isRequired,
  screenReaderLabel: string.isRequired, // basically a summarizing title
  shouldCloseOnOverlayClick: bool,
};

Modal.defaultProps = {
  className: undefined,
  isOpen: false,
  shouldCloseOnOverlayClick: true,
};

function Modal({
  children,
  className,
  isOpen,
  onRequestClose,
  screenReaderLabel,
  shouldCloseOnOverlayClick,
}) {
  if (isOpen) {
    gtag.modalView(screenReaderLabel);
  }

  return (
    <ReactModal
      portalClassName={ModalStyles.Modal}
      className={classNames(CardStyles.Card, className)}
      contentLabel={screenReaderLabel}
      isOpen={isOpen}
      onRequestClose={onRequestClose}
      shouldCloseOnOverlayClick={shouldCloseOnOverlayClick}
    >
      <CloseButton onClick={onRequestClose} />
      <div className={ModalStyles.scrollableContainer}>{children}</div>
    </ReactModal>
  );
}

export default Modal;
