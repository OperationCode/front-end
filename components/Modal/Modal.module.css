.Modal {
  position: relative;
}

.scrollableContainer {
  margin-top: 2.5rem;
  overflow: auto;
  max-height: 75vh;
}

/* React-Modal Overrides
 * Remove scroll on the body when react-modal is open */
/* stylelint-disable */
.Modal :global(.ReactModal__Html--open),
.Modal :global(.ReactModal__Body--open) {
  height: 100%;
  overflow: hidden;
}

.Modal :global(.ReactModal__Overlay) {
  align-items: center;
  display: flex;
  justify-content: center;
}

.Modal :global(.ReactModal__Content) {
  width: 80%;
}
