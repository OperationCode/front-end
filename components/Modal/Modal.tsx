import classNames from 'classnames';
import * as Dialog from '@radix-ui/react-dialog';
import { gtag } from 'common/utils/thirdParty/gtag';
import CloseButton from 'components/Buttons/CloseButton/CloseButton';
import { MODAL_CONTENT, MODAL_OVERLAY } from 'common/constants/testIDs';

export type ModalPropsType = {
  /**
   * Content to be rendered in the modal.
   */
  children: React.ReactNode;
  /**
   * Function that is called when the user clicks the close button.
   */
  onRequestClose: (arg1: any) => void;
  /**
   * Applies a label for the screen reader.
   */
  screenReaderLabel: string;
  /**
   * Applies style classes to the wrapping div.
   */
  className?: string;
  /**
   * Sets if the modal is open an visible (or not)
   * @default false
   */
  isOpen?: boolean;
  /**
   * Sets if the modal can be closed by the user
   * @default true
   */
  canClose?: boolean;
  /**
   * Applies style classes to the child content.
   */
  childrenClassName?: string;
  /**
   * Applies classNames to the overlay.
   */
  overlayClassName?: string;
};

function Modal({
  children,
  className,
  isOpen = false,
  onRequestClose,
  screenReaderLabel,
  canClose = true,
  childrenClassName,
}: ModalPropsType) {
  if (isOpen) {
    gtag.modalView(screenReaderLabel);
  }

  const portalContainer =
    typeof window !== 'undefined'
      ? (document.querySelector('#__next') as HTMLElement) ?? undefined
      : undefined;

  return (
    <Dialog.Root defaultOpen={false} open={isOpen}>
      <Dialog.Portal container={portalContainer}>
        <Dialog.Overlay
          className="inset-0 fixed bg-white/50 z-[2]"
          onClick={canClose ? onRequestClose : undefined}
          data-testid={MODAL_OVERLAY}
        >
          {canClose && (
            <Dialog.Close asChild>
              <CloseButton theme="secondary" onClick={onRequestClose} />
            </Dialog.Close>
          )}
        </Dialog.Overlay>

        <Dialog.Content
          className={classNames(
            'outline-none',
            'bg-white',
            'text-themeSecondary',
            'flex',
            'flex-col',
            'flex-nowrap',
            'm-4',
            'min-h-[100px]',
            'min-w-[100px]',
            'p-6',
            'fixed',
            'items-center',
            'justify-center',
            'overflow-hidden',
            'w-[80%]',
            'top-1/2',
            'left-1/2',
            '-translate-x-1/2',
            '-translate-y-1/2',
            'z-[2]',
            'shadow-md',
            className,
          )}
        >
          <div className={childrenClassName} data-testid={MODAL_CONTENT}>
            {children}
          </div>
        </Dialog.Content>
      </Dialog.Portal>
    </Dialog.Root>
  );
}

export default Modal;
