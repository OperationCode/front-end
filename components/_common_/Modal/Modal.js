import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import ReactModal from 'react-modal';
import ReactGA from 'react-ga';
<<<<<<< HEAD
import CloseButton from 'components/_common_/CloseButton/CloseButton';
=======
>>>>>>> 1e3818fe1ed61deeda96881e0e7a718d44c1dd4e
import CardStyles from 'components/_common_/Card/Card.css';
import ModalStyles from './Modal.css';

Modal.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
<<<<<<< HEAD
  hasCloseIcon: PropTypes.bool,
  isOpen: PropTypes.bool,
  onRequestClose: PropTypes.func,
=======
  isOpen: PropTypes.bool,
  onRequestClose: PropTypes.func.isRequired,
>>>>>>> 1e3818fe1ed61deeda96881e0e7a718d44c1dd4e
  screenReaderLabel: PropTypes.string.isRequired, // basically a summarizing title
  shouldCloseOnOverlayClick: PropTypes.bool,
};

Modal.defaultProps = {
<<<<<<< HEAD
  className: '',
  hasCloseIcon: true,
  isOpen: false,
  onRequestClose: undefined,
=======
  className: undefined,
  isOpen: false,
>>>>>>> 1e3818fe1ed61deeda96881e0e7a718d44c1dd4e
  shouldCloseOnOverlayClick: true,
};

function Modal({
  children,
  className,
<<<<<<< HEAD
  hasCloseIcon,
=======
>>>>>>> 1e3818fe1ed61deeda96881e0e7a718d44c1dd4e
  isOpen,
  onRequestClose,
  screenReaderLabel,
  shouldCloseOnOverlayClick,
}) {
  if (process.env.NODE_ENV === 'production' && process.env.GOOGLE_ANALYTICS_KEY) {
    ReactGA.modalview(screenReaderLabel);
  }

  return (
    <ReactModal
      className={classNames(CardStyles.Card, ModalStyles.ModalCard, className)}
      contentLabel={screenReaderLabel}
      isOpen={isOpen}
      onRequestClose={onRequestClose}
      shouldCloseOnOverlayClick={shouldCloseOnOverlayClick}
    >
<<<<<<< HEAD
      {hasCloseIcon && <CloseButton onClick={onRequestClose} />}

=======
>>>>>>> 1e3818fe1ed61deeda96881e0e7a718d44c1dd4e
      <div className={ModalStyles.scrollableContainer}>{children}</div>
    </ReactModal>
  );
}

export default Modal;
