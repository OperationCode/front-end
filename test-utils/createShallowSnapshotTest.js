import ShallowRenderer from 'react-test-renderer/shallow';

/**
 * Used to create a snapshot test off of surface leaves of the tree.
 * Useful for regression tests on containers, components utilizing something that requires large
 * mocking, or very large components.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const shallowRenderer = new ShallowRenderer();
  const tree = shallowRenderer.render(Component);

  expect(tree).toMatchSnapshot();
};
