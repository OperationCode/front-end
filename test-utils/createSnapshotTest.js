/* eslint-env jest */
import renderer from 'react-test-renderer';

/**
 * Used to create fully-rendered snapshot test.
 * Useful for reusable component smoke tests.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const tree = renderer.create(Component).toJSON();
  expect(tree).toMatchSnapshot();
};
