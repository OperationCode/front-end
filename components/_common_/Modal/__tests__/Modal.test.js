/* eslint-env jest */
import React from 'react';
import { shallow } from 'enzyme';
import ReactGA from 'react-ga';
import createSnapshotTest from 'test-utils/createSnapshotTest';
<<<<<<< HEAD
import createShallowSnapshotTest from 'test-utils/createShallowSnapshotTest';
=======
>>>>>>> 1e3818fe1ed61deeda96881e0e7a718d44c1dd4e

import Modal from '../Modal';

describe('Modal', () => {
<<<<<<< HEAD
  test('should render with just required props passed', () => {
    createSnapshotTest(
      <Modal screenReaderLabel="Test" onRequestClose={() => {}}>
=======
  it('should render with required props', () => {
    createSnapshotTest(
      <Modal screenReaderLabel="Test" onRequestClose={jest.fn()}>
>>>>>>> 1e3818fe1ed61deeda96881e0e7a718d44c1dd4e
        Test
      </Modal>,
    );
  });

<<<<<<< HEAD
  test('should render properly with some props assigned and not being open', () => {
    createSnapshotTest(
      <Modal
        className="test-class"
        onRequestClose={() => {}}
        screenReaderLabel="Test"
        shouldCloseOnOverlayClick
      >
        Test
      </Modal>,
    );
  });

  test('should render properly with some props assigned and being open', () => {
    createShallowSnapshotTest(
      <Modal
        className="test-class"
        isOpen
        onRequestClose={() => {}}
        screenReaderLabel="Test"
        shouldCloseOnOverlayClick
=======
  it('should render with many props assigned', () => {
    createSnapshotTest(
      <Modal
        className="test-class"
        isOpen
        onRequestClose={jest.fn()}
        screenReaderLabel="Test"
        shouldCloseOnOverlayClick={false}
>>>>>>> 1e3818fe1ed61deeda96881e0e7a718d44c1dd4e
      >
        Test
      </Modal>,
    );
  });

<<<<<<< HEAD
  test('should call ReactGA when in prod environment', () => {
=======
  it('should call ReactGA when in prod environment', () => {
>>>>>>> 1e3818fe1ed61deeda96881e0e7a718d44c1dd4e
    ReactGA.initialize('foo', { testMode: true });

    process.env.NODE_ENV = 'production';
    process.env.GOOGLE_ANALYTICS_KEY = 'debug';

    const props = {
<<<<<<< HEAD
      hasCloseIcon: false,
=======
>>>>>>> 1e3818fe1ed61deeda96881e0e7a718d44c1dd4e
      onRequestClose: () => {},
      screenReaderLabel: 'Test',
    };

    shallow(<Modal {...props}>Testing</Modal>);

    expect(ReactGA.testModeAPI.calls).toContainEqual(['send', 'pageview', '/modal/Test']);
  });
});
