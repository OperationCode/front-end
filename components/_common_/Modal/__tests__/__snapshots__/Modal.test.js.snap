// Jest Snapshot v1, https://goo.gl/fbAQLP

<<<<<<< HEAD
exports[`Modal should render properly with some props assigned and being open 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="Card ModalCard test-class"
  closeTimeoutMS={0}
  contentLabel="Test"
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[MockFunction]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <CloseButton
    disabled={false}
    onClick={[Function]}
  />
  <div
    className="scrollableContainer"
  >
    Test
  </div>
</Modal>
`;

exports[`Modal should render properly with some props assigned and not being open 1`] = `null`;

exports[`Modal should render with just required props passed 1`] = `null`;
=======
exports[`Modal should render with many props assigned 1`] = `null`;

exports[`Modal should render with required props 1`] = `null`;
>>>>>>> 1e3818fe1ed61deeda96881e0e7a718d44c1dd4e
